include/seqan3/core/algorithm/configuration_element_parallel_mode.hpp

```diff
@@ -25,9 +26,32 @@ namespace seqan3::detail
  * This type is used to enable the parallel mode of the algorithms.
  */
 template <typename wrapped_config_id_t>
-struct parallel_mode : public pipeable_config_element<parallel_mode<wrapped_config_id_t>, uint32_t>
+class parallel_mode : public pipeable_config_element<parallel_mode<wrapped_config_id_t>>
 {
-    //!\brief Internal id to check for consistent configuration settings.
+public:
+    /*!\name Constructors, assignment and destructor
+     * \{
+     */
+    parallel_mode() = default; //!< Defaulted.
+    parallel_mode(parallel_mode const &) = default; //!< Defaulted.
+    parallel_mode(parallel_mode &&) noexcept = default; //!< Defaulted.
+    parallel_mode & operator=(parallel_mode const &) = default; //!< Defaulted.
+    parallel_mode & operator=(parallel_mode &&) noexcept = default; //!< Defaulted.
+    ~parallel_mode() = default; //!< Defaulted.
+
+    /*!\brief Sets the number of threads for the parallel configuration element.
+     * \param[in] thread_c The maximum number of threads to be used by the algorithm.
+     */
+    explicit parallel_mode(uint32_t thread_c) noexcept : thread_count{thread_c}
+    {}
+    //!\}
+
+    //!\brief The maximum number of threads the algorithm can use.
+    uint32_t thread_count{1u};
```

Comment:
**Question:** Do we want `std::thread::hardware_concurrency` as a default here?

Reply:
hmm good question. I would vote for yes. I'll ask @seqan/core